CMakeLists.txt

```diff
@@ -474,6 +490,13 @@ endif()
 FILE(GLOB_RECURSE SOURCE "src/*.cc" "src/*.h" "include/*.h")
 FILE(GLOB_RECURSE CUDA "src/*.cu" "src/*.cuh")
 
+if (USE_INTGEMM)
+  list(APPEND SOURCE "3rdparty/intgemm/intgemm.cc")
+else()
+  FILE(GLOB_RECURSE INTGEMM_OPERATOR_SOURCE "src/operator/contrib/intgemm/*.cc" "src/operator/contrib/intgemm/*.h")
+  list(REMOVE_ITEM SOURCE ${INTGEMM_OPERATOR_SOURCE})
+endif()
```

Comment:
`add_subdirectory` makes available all exported targets. You should only need a call à la `target_link_libraries(mxnet PRIVATE intgemm)`

Reply:
Think this was meant to be a comment on `add_subdirectory`?  The minimal build needs `intgemm.cc` compiled in and these compilation tests that generate a header file: https://github.com/kpu/intgemm/blob/0f05c3ebd037eacdf8cff165736fea2b0d125023/CMakeLists.txt#L20-L37

Not sure how to make `cmake` run the compilation tests without `add_subdirectory`?  I can do `add_subdirectory( 3rdparty/intgemm EXCLUDE_FROM_ALL)` if that helps.

Reply:
Added `EXCLUDE_FROM_ALL`

Reply:
Sorry for being unclear in the previous comment. What I mean is, that if you use `add_subdirectory`, there is no need for adding `INTGEMM_OPERATOR_SOURCE` manually to the list of source files of the mxnet library. Instead the `CMakeLists.txt` should declare a target, which encapsulates the set of source and header files (and triggers the compilation tests). Then, in the mxnet CMakeLists.txt you only need `add_subdirectory` and `target_link_libraries(mxnet PRIVATE intgemm)`

Reply:
33ad78208c911a5e9d770416266b0d3510767213 now uses `target_link_libraries(mxnet PRIVATE intgemm)`

Regarding `INTGEMM_OPERATOR_SOURCE`, that's not adding stuff, it's removing the MXNet operator implementations (which were caught by the glob) if the underlying library isn't compiled in.

Reply:
I take your :+1:  emoji to mean this part is resolved?